test/dump/relocations-indirect-calls-and-table-elements.txt

```diff
@@ -0,0 +1,70 @@
+;;; TOOL: run-objdump
+;;; ARGS0: -r
+;;; ARGS1: -x
+(module
+  (func $add (param i32)(param i32)(result i32)
+    get_local 0
+    get_local 1
+    i32.add
+    return)
+  (table 1 anyfunc)
+  (elem (i32.const 0) $add)
+  (func $run
+    i32.const 4444
+    i32.const 5555
+    i32.const 0
+    call_indirect (param i32)(param i32)(result i32)
+    drop)
+  (export "run" (func $run)))
+(;; STDOUT ;;;
+
+relocations-indirect-calls-and-table-elements.wasm:	file format wasm 0x1
+
+Section Details:
+
+Type[2]:
+ - type[0] (i32, i32) -> i32
+ - type[1] () -> nil
+Function[2]:
+ - func[0] sig=0 <$add>
+ - func[1] sig=1 <run>
+Table[1]:
+ - table[0] type=funcref initial=1
+Export[1]:
+ - func[1] <run> -> "run"
+Elem[1]:
+ - segment[0] flags=0 table=0 count=1 - init i32=0
+  - elem[0] = func[0] <$add>
+Code[2]:
+ - func[0] size=8 <$add>
+ - func[1] size=18 <run>
+Custom:
+ - name: "linking"
+  - symbol table [count=1]
+   - 0: F <$add> func=0 binding=global vis=default
+Custom:
+ - name: "reloc.Elem"
+  - relocations for section: 4 (Elem) [1]
+   - R_WASM_FUNCTION_INDEX_LEB offset=0x000006(file=0x000030) symbol=0 <$add>
```

Comment:
Interesting.   In llvm we don't currently use any relocations in the Elem section,  This is because the linker creates the elem section from scratch based on the functions that get called indirectly.

So I don't think we need this.

Maybe just call this new test `relocations_type_index.txt`?  Or just add to the existing `relocations.txt`

Reply:
Yeah I looks like we only currently support relocations in DATA, CODE, and custom sections:

https://github.com/llvm/llvm-project/blob/ca688ae4975932f0566b113dad10a66db852259e/lld/wasm/Writer.cpp#L168

Reply:
The elem reloc section was already being created, I didn't introduce it. Does the linker correctly create the elements with relocated indices? If so, I'll look into removing the whole section. And yeah I'll rename the test.

Reply:
Ah, reloc sections are automatically created based on the name of the section in which a reloc is added. So we would just remove the relocs in the elem section then to remove the elem reloc section.

Reply:
Yup!